Dynatrace/dynatrace-operator
# EdgeConnect config enhancements for k8s automation.

## Description

This PR adds an extra service account certificate and adds a `secrets` section to the config to EC config.

## How can this be tested?

`make go/test`

Deploy EC to your cluster:
```
apiVersion: dynatrace.com/v1alpha1
kind: EdgeConnect
metadata:
  name: e2e-andrii
  namespace: dynatrace
spec:
  apiServer: "<tenant>.dev.apps.example.com"
  replicas: 1
  oauth:
    clientSecret: edge-connect-client-secret-<tenant>
    endpoint: https://sso-dev.example.com/sso/oauth2/token
    resource: urn:dtenvironment:<tenannt>
```

and connect using debug pod and check yaml and EC logs.

edgeConnect.yaml:
```
k get secrets e2e-andrii-ec-yaml -o jsonpath="{.data['edgeConnect\.yaml']}" | base64 -d
name: e2e-andrii
api_endpoint_host: ******.dev.apps.******.com
oauth:
    endpoint: https://sso-dev.******.com/sso/oauth2/token
    client_id: ******
    client_secret: ******.******.******
    resource: urn:dtenvironment:******
root_certificate_paths:
    - /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
secrets:
    - name: K8S_SERVICE_ACCOUNT_TOKEN
      token: ******.******.******
      from_file: /var/run/secrets/kubernetes.io/serviceaccount/token
      restrict_hosts_to:
        - kubernetes.default.svc
```
